lib/api/model/model.dart

```diff
@@ -698,3 +710,82 @@ class DmMessage extends Message {
   @override
   Map<String, dynamic> toJson() => _$DmMessageToJson(this);
 }
+
+enum MessageEditState {
+  none,
+  edited,
+  moved;
+
+  // Adapted from the shared code:
+  //   https://github.com/zulip/zulip/blob/1fac99733/web/shared/src/resolved_topic.ts
+  // The canonical resolved-topic prefix.
+  static const String _resolvedTopicPrefix = '✔ ';
+
+  static bool _isResolved(String topic) {
+    return topic.startsWith(_resolvedTopicPrefix);
+  }
+
+  /// Whether two topics are equal, ignoring any resolved-topic prefix.
+  ///
+  /// When a topic is resolved, the clients agree on adding a ✔ prefix to the
+  /// topic string. Topics whose only difference is the ✔ prefix are considered
+  /// the same. This helper can be helpful when checking if a message has been
+  /// moved.
+  ///
+  /// TODO(#744) Extract to its own home when supporting "mark as resolve".
+  static bool sameTopicModuloResolve(String topic, String prevTopic) {
```

Comment:
nit: not something relevant for trying to use the function, so belongs as an implementation comment rather than dartdoc:
```suggestion
  /// moved.
  //
  // TODO(#744) Extract to its own home when supporting "mark as resolve".
  static bool sameTopicModuloResolve(String topic, String prevTopic) {
```